src/atomate2/vasp/sets/base.py

```diff
@@ -638,21 +637,30 @@ def _get_incar(
         if self.use_structure_charge:
             incar["NELECT"] = self.get_nelect(structure)
 
-        # handle kspacing
-        _set_kspacing(
-            incar,
-            incar_settings,
-            self.user_incar_settings,
-            self.auto_kspacing,
-            bandgap,
-            kpoints,
-            previous_incar is None,
-        )
-
         # handle auto ISPIN
         if ispin is not None and "ISPIN" not in self.user_incar_settings:
             incar["ISPIN"] = ispin
 
+        if self.auto_ismear:
+            if bandgap is None:
+                # don't know if we are a metal or insulator so set ISMEAR and SIGMA to
+                # be safe with the most general settings
+                incar.update({"SIGMA": 0.2, "ISMEAR": 0})
+            elif bandgap == 0:
+                incar.update({"SIGMA": 0.2, "ISMEAR": 2})  # metal
+            else:
+                incar.update({"ISMEAR": -5, "SIGMA": 0.05})  # insulator
+
+        if self.auto_lreal:
+            incar.update({"LREAL": _get_recommended_lreal(structure)})
+
+        if kpoints is not None:
```

Comment:
@jmmshn All of the new test failures in #412  are cases where ISMEAR is changed from -5 to 0. Might this be the culprit?

Reply:
@matthewkuner Looks like things got fixed by @utf.   Either way the LREAL behavior is not really checke by too many things so I doubt it was the cause.